src/test/java/plugins/WarningsPluginTest.java

```diff
@@ -86,6 +82,289 @@
+        warningsPublisher.addConsoleScanner(JAVA_ID);
+
+        job.save();
+
+        Build build = buildSuccessfulJob(job);
+
+        assertThatActionExists(job, build, "Java Warnings");
+
+        WarningsAction action = createJavaResultAction(build);
+
+        action.open();
+
+        SortedMap<String, String> map = action.getDetailsTabContents();
+        Set<Map.Entry<String, String>> set = map.entrySet();
+
+        assertThat("Assert the proper detail count.", set.size(), is(3));
+
+        // String is a comparable -> keys are sorted.
+        String firstKey = map.firstKey();
+        String firstValue = map.get(firstKey);
+        String[] firstHeader = firstKey.split(",");
+        assertThat("Assert the proper header array length." , firstHeader.length, is(3));
+        assertThat("Assert the proper file and line." , firstHeader[0].trim(), is("WarningMain.java:10"));
+        assertThat("Assert the proper parser." , firstHeader[1].trim(), is("Java Compiler (javac)"));
+        assertThat("Assert the proper warning priority." , firstHeader[2].trim(), is("Priority: Normal"));
+        assertThat("Assert the proper detail text" , firstValue.trim(), is("redundant cast to TextClass"));
+
+        // removing tested entry from map.
+        map.remove(firstKey);
+
+        String secondKey = map.firstKey();
+        String secondValue = map.get(secondKey);
+        String[] secondHeader = secondKey.split(",");
+        assertThat("Assert the proper header array length." , secondHeader.length, is(3));
+        assertThat("Assert the proper file and line." , secondHeader[0].trim(), is("WarningMain.java:11"));
+        assertThat("Assert the proper parser." , secondHeader[1].trim(), is("Java Compiler (javac)"));
+        assertThat("Assert the proper warning priority." , secondHeader[2].trim(), is("Priority: Normal"));
+        assertThat("Assert the proper detail text" , secondValue.trim(), is("redundant cast to TextClass"));
+
+        // removing tested entry from map.
+        map.remove(secondKey);
+
+        String thirdKey = map.firstKey();
+        String thirdValue = map.get(thirdKey);
+        String[] thirdHeader = thirdKey.split(",");
+        assertThat("Assert the proper header array length." , thirdHeader.length, is(3));
+        assertThat("Assert the proper file and line." , thirdHeader[0].trim(), is("WarningMain.java:14"));
+        assertThat("Assert the proper parser." , thirdHeader[1].trim(), is("Java Compiler (javac)"));
+        assertThat("Assert the proper warning priority." , thirdHeader[2].trim(), is("Priority: Normal"));
+        assertThat("Assert the proper detail text" , thirdValue.trim(), is("division by zero"));
+
+    }
+
+    @WithDocker
+    @Test
+    public void detailsTabContentWithOneWarningTest() throws ExecutionException, InterruptedException {
+        // TODO maybe refactoring is necessary - removing duplicates
```

Comment:
detailsTabContentWithOneWarningTest: This tests the table with one Warning.
detailsTabContentTest: This tests the table with more than one Warning. (3 warnings)

Tests with 0 warnings are unnecessary, because in that case there is no table to check.